crates/astria-sequencer/src/fees/mod.rs

```diff
@@ -108,381 +79,379 @@ impl Fee {
 }
 
 #[async_trait::async_trait]
-impl FeeHandler for Transfer {
-    type FeeComponents = TransferFeeComponents;
+pub(crate) trait FeeHandler: Send {
+    /// The Pascal-case type name, e.g. `RollupDataSubmission`.
+    // NOTE: We only require this function due to `IbcRelay` not implementing `Protobuf`.
+    fn name() -> &'static str;
+
+    /// The full name including the protobuf package, e.g.
+    /// `astria.protocol.transaction.v1.RollupDataSubmission`.
+    // NOTE: We only require this function due to `IbcRelay` not implementing `Protobuf`.
+    fn full_name() -> String;
+
+    /// The snake-case type name, e.g. `rollup_data_submission`.
+    fn snake_case_name() -> &'static str;
```

Comment:
While it makes perfect sense that we need this functionality for the `IbcRelay` action, I think it is a bit odd to include this in `FeeHandler`, especially when these methods are applicable to many other things than fees. Perhaps we could do something like make a new trait `ProstNameExt` or `NameExt`, implement it for all actions which implement protobuf and add our own implementation for `IbcRelay`, then require it for `FeeHandler`? Just a thought, and may also be out of the scope of this PR.

Reply:
I guess I'd rather keep this functionality as restricted as possible until we have a concrete use-case for adding it more widely.  The core lib is already pretty busy, and I'd rather avoid adding another trait there for it to only be consumed in one specific place in the sequencer.